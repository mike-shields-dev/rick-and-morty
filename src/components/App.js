import React, { useState } from "react";
import styles from "../styles/app.module.css";
import Controls from "./Controls";

const App = () => {
  const [status, setStatus] = useState("all");
  const [page, setPage] = useState(1);
  console.log({ status, page });

  return (
    <div className={styles.app}>
      <header className={styles.app__header}>
        <h1 className={styles.app__heading}>Rick and Morty</h1>
        <p className={styles.app__tagline}>
          {`"I'm a scientist because I invent, transform, create, and destroy for
          a living, and when I don't like something about the world, I change
          it."`}
        </p>
      </header>
      <main className={styles.app__main}>
        <Controls {...{ status, setStatus, page, setPage }} />
      </main>
    </div>
  );
};

export default App;
