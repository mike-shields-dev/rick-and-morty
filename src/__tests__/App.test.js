import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

const title = "Rick and Morty";
const tagline = `"I'm a scientist because I invent, transform, create, and destroy for a living, and when I don't like something about the world, I change it."`;

const renderApp = () => render(<App />);

describe(`App`, () => {
  it(`matches snapshot`, () => {
    const { asFragment } = renderApp();

    expect(asFragment()).toMatchSnapshot();
  });

  it(`displays a heading with the text ${title}`, () => {
    renderApp();

    expect(screen.getByText(title)).toBeInTheDocument();
  });

  it(`displays a tagline with the text: ${tagline}`, () => {
    renderApp();

    expect(screen.getByText(tagline)).toBeInTheDocument();
  });
});
