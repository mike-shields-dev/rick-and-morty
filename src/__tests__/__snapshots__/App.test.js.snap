// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header
      class="app__header"
    >
      <h1
        class="app__heading"
      >
        Rick and Morty
      </h1>
      <p
        class="app__tagline"
      >
        "I'm a scientist because I invent, transform, create, and destroy for
          a living, and when I don't like something about the world, I change
          it."
      </p>
    </header>
    <main
      class="app__main"
    >
      <div
        class="controls"
      >
        <div
          class="controls__panel left"
        >
          <fieldset
            class="status-filter"
          >
            <legend>
              Status: all
            </legend>
            <div
              class="status-filter__inputs"
            >
              <label
                class="status-filter__label"
                for="alive"
              >
                alive
                <input
                  class="status-filter__input"
                  id="alive"
                  name="status"
                  type="radio"
                  value="alive"
                />
              </label>
              <label
                class="status-filter__label"
                for="dead"
              >
                dead
                <input
                  class="status-filter__input"
                  id="dead"
                  name="status"
                  type="radio"
                  value="dead"
                />
              </label>
              <label
                class="status-filter__label"
                for="all"
              >
                all
                <input
                  checked=""
                  class="status-filter__input"
                  id="all"
                  name="status"
                  type="radio"
                  value="all"
                />
              </label>
            </div>
          </fieldset>
        </div>
        <div
          class="controls__panel right"
        >
          <div
            class="page-counter"
          >
            <label
              for="page"
            >
              Page
            </label>
            <div
              class="page-counter__input-wrap"
            >
              <button
                class="page-counter__button page-counter__button--dec "
                disabled=""
                type="button"
                value="-1"
              >
                -
              </button>
              <button
                class="page-counter__button page-counter__button--inc"
                type="button"
                value="1"
              >
                +
              </button>
              <input
                class="page-counter__input"
                id="page"
                type="number"
                value="1"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
